src/main/java/cz/tstrecha/timetracker/repository/TaskRepository.java

```diff
@@ -1,19 +1,58 @@
 package cz.tstrecha.timetracker.repository;
 
+import cz.tstrecha.timetracker.constant.TaskField;
+import cz.tstrecha.timetracker.constant.TaskStatus;
+import cz.tstrecha.timetracker.dto.LoggedUser;
+import cz.tstrecha.timetracker.dto.TaskFilter;
 import cz.tstrecha.timetracker.repository.entity.TaskEntity;
+import cz.tstrecha.timetracker.repository.entity.TaskEntity_;
 import cz.tstrecha.timetracker.repository.entity.UserEntity;
+import cz.tstrecha.timetracker.util.FilterUtils;
+import jakarta.persistence.criteria.Predicate;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.domain.Specification;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
 import org.springframework.data.jpa.repository.Query;
 
+import java.util.ArrayList;
 import java.util.List;
 import java.util.Optional;
 
-public interface TaskRepository extends JpaRepository<TaskEntity, Long> {
+public interface TaskRepository extends JpaRepository<TaskEntity, Long>, JpaSpecificationExecutor<TaskEntity> {
 
     @Query(value = "SELECT * FROM task WHERE (name_simple ILIKE :query OR custom_id ILIKE :query" +
             " OR CONCAT(custom_id, CONCAT(' - ', name_simple)) ILIKE :query ) " +
             "AND user_id = :loggedUserId LIMIT :resultLimit", nativeQuery = true)
     List<TaskEntity> searchForTasks(String query, Long loggedUserId, Long resultLimit);
 
     Optional<TaskEntity> findByIdAndUser(Long id, UserEntity user);
-}
+
+    default List<TaskEntity> findByFilter(TaskFilter taskFilter, Pageable pageable, LoggedUser loggedUser){
+        return findAll((Specification<TaskEntity>) (root, query, cb) -> {
+            query.distinct(true);
+
+            List<Predicate> predicates = new ArrayList<>();
+            predicates.add(cb.equal(root.get(TaskEntity_.USER), loggedUser.getUserEntity()));
+
+            for (TaskField taskField : taskFilter.getFieldFilters().keySet()) {
+                switch (taskField) {
+                    case ID -> predicates.add(cb.equal(root.get(TaskEntity_.ID), taskFilter.getFieldFilters().get(taskField)));
```

Comment:
DRY Rule => Youre calling`taskFilter.getFieldFilters().get(taskField))` on every single line -> you can extract it into var and dont repeat yourself (DRY)